addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1313,6 +1337,31 @@ export class OdooEditor extends EventTarget {
         if (!range) return;
         let start = range.startContainer;
         let end = range.endContainer;
+        // Expand the range to fully include all contentEditable=False elements.
+        let commonAncestorContainer = range.commonAncestorContainer;
+        if (commonAncestorContainer !== this.editable && !this.editable.contains(commonAncestorContainer)) {
+            commonAncestorContainer = this.editable;
+        }
+        const startUneditable = this._getFurthestUneditableParent(start, commonAncestorContainer);
+        if (startUneditable) {
+            let leaf = previousLeaf(startUneditable);
+            if (leaf) {
+                range.setStart(leaf, nodeSize(leaf));
+            } else {
+                range.setStart(commonAncestorContainer, 0);
+            }
+            start = range.startContainer;
```

Comment:
If you set `let start = range.startContainer;` and `let end = range.endContainer;` after the `if (endUneditable) {`, you don't need that instruction nor `end = range.endContainer;`